File: 2/code.go
package main

import (
  "fmt"
  "bufio"
  "log"
  "os"
  "strings"
  "strconv"
)

var (
  // 1, 2 or 3
  part int = 1
  sample bool = false
)

func helper(val int) int {
  if sample {
    return 1
  } else {
    return 0
  }
}

func counter(w string, ch string) (c int) {
  c = 0
  for _, v := range w {
    if string(v) == (ch) {
      c += 1
    }
  }
  return
}

func partone(s []string) (ans int) {
  ans = 0
  //var i []string
  var char string
  //fmt.Println(s)
  for _, v := range s {
    //fmt.Println("Part two:", v)
    i := strings.Split(v, " ")
    j := strings.Split(i[0], "-")
    low, _ := strconv.Atoi(j[0])
    high, _ := strconv.Atoi(j[1])
    char = string(i[1][0])
    word := string(i[2])
    c := counter(word, char)
    if (low <= c) && (c <= high) {
      ans += 1
    }
    //fmt.Println(i, low, high, char, word)
  }

  fmt.Println("Part one:", ans)
  return ans
}

func parttwo(s []string) (ans int) {
  ans = 2
  fmt.Println("Part two:", ans)
  return ans
}

func main() {
  var filename string = "./input.txt"
  if sample {
    filename = "./sample.txt"
  }
  file, err := os.Open(filename)
  if err != nil {
    log.Fatal(err)
  }
  defer file.Close()
  var s []string
  scanner := bufio.NewScanner(file)
  for scanner.Scan() {
    s = append(s, scanner.Text())
    //fmt.Println(scanner.Text())
  }
  //fmt.Println(scanner)
  if err := scanner.Err(); err != nil {
    log.Fatal(err)
  }
  //var fuelsum int = 0
  if part == 1 {
    partone(s)
  } else if part == 2 {
    parttwo(s)
  } else {
    partone(s)
    parttwo(s)
  }
}

File: 2/code.py
#!/usr/bin/python3

#from collections import Counter
#import re
#import os
import time
#from collections import defaultdict
#from collections import deque
'''     #######     '''

date = 2
dev = 0 # extra prints
part = 1 # 1,2, or 3 for both
samp = 0 # 0 or 1

'''     #######     '''

def count(c, word):
    j = 0
    for i in word:
        if i == c:
            j += 1
    return j

def day(te):
    valid = 0
    for t in te:
        t = (t.replace('-',' ').replace(':','').split())
        #print(t)
        low = int(t[0])
        high = int(t[1])
        char = t[2]
        word = t[3]
        times = count(char, word)
        if (low <= times) and (times <= high):
            valid += 1
    return valid

def day2(te):
    valid = 0
    for t in te:
        t = (t.replace('-',' ').replace(':','').split())
        #print(t)
        low = int(t[0]) - 1
        high = int(t[1]) - 1
        char = t[2]
        word = t[3]
        times = 0
        #times = count(char, word)
        if word[low] == char:
            times += 1
        if word[high] == char:
            times += 1
        if times == 1:
            valid += 1
    return valid

'''     #######     '''

time0 = time.time()

if samp == 1:
    filename = "/sample.txt"
else:
    filename = "/input.txt"

try:
    with open(str(date) + filename,"r") as f:
        t = f.readlines()
except FileNotFoundError:
    with open("." + filename,"r") as f:
        t = f.readlines()

t = [(x.strip().replace('  ',' ')) for x in t]

if part == 1:
    print("Part 1: ", day(t))
elif part == 2:
    print("Part 2: ", day2(t))
elif part == 3:
    #run both
    print("Part 1: ", day(t))
    print("Part 2: ", day2(t))

tdif = time.time() - time0
print("Elapsed time: {:.4f} s".format(tdif))
